### PR TITLE
Supports cuda version BorderAlign module

### PR DESCRIPTION
**Pull request by v-qjqs**
Support for cuda/gpu version `BorderAlign` module, for more details please refer to the paper: [BorderDet: Border Feature for Dense Object Detection](https://arxiv.org/abs/2007.11056).

### CODE REVIEW

File: mmcv/ops/border_align.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        return grad_input, None, None
+
+
+border_align = BorderAlignFunction.apply
+
+
+class BorderAlign(nn.Module):
+    r"""Border align pooling layer.
+
+    Applies border_align over the input feature based on predicted bboxes.
```
Comment (ZwwWayne) [inline mmcv/ops/border_align.py:64]: Should we indicate which file is this module modified from?

File: mmcv/ops/csrc/border_align_cuda_kernel.cuh [hunk 2]
```diff
@@ -0,0 +1,193 @@
+#ifndef BORDER_ALIGN_CUDA_KERNEL_CUH
```
Comment (ZwwWayne) [inline mmcv/ops/csrc/border_align_cuda_kernel.cuh:1]: Should we indicate which file is this module modified from in the header?

File: mmcv/ops/border_align.py [hunk 3]
```diff
@@ -8,6 +8,9 @@
 ext_module = ext_loader.load_ext(
     '_ext', ['border_align_forward', 'border_align_backward'])
 
+# modified from
```
Comment (ZwwWayne) [inline mmcv/ops/border_align.py:11]: Need to be put in the top of the file.

File: mmcv/ops/csrc/border_align_cuda_kernel.cuh [hunk 4]
```diff
@@ -12,6 +12,12 @@
 #endif  // MMCV_USE_PARROTS
 #endif  // MMCV_WITH_TRT
 
+// modified from
```
Comment (ZwwWayne) [inline mmcv/ops/csrc/border_align_cuda_kernel.cuh:15]: put it to the top of the file.

**Review by ZwwWayne — APPROVED** [review]

**Comment by ZwwWayne** [comment]
Can be merged after fixing lint issue.

**Review by ZwwWayne — APPROVED** [review]

**Comment by ZwwWayne** [comment]
@v-qjqs can be merged after resolving conflicts.

**Comment by v-qjqs** [comment]
> @v-qjqs can be merged after resolving conflicts.

fixed.

**Pull request merged**